src/os_win/os_futex.c

```diff
@@ -0,0 +1,49 @@
+/*-
+ * Copyright (c) 2024-present MongoDB, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include "wt_internal.h"
+
+/*
+ * __wt_futex_wait --
+ *     Wait on the futex. The timeout is in microseconds and MUST be greater than zero.
+ */
+int
+__wt_futex_wait(
+  volatile WT_FUTEX_WORD *addr, WT_FUTEX_WORD expected, time_t usec, WT_FUTEX_WORD *wake_valp)
+{
+    DWORD msec;
+    DWORD windows_error;
+    bool retval;
+
+    msec = (DWORD)WT_MAX(1, usec / 1000);
```

Comment:
May as well have the same assertions as the Linux version? (e.g. that `usec > 0` and `wake_val` is sane)

Reply:
Added usec check. There is no restriction on `wake_val`.

Reply:
Sorry, I meant `wake` rather than `wake_val` - as in the assertion that it's `FUTEX_WAIT_ONE` or `WAIT_ALL`.

Reply:
yeah okay, i think that is already present below in wake

Reply:
So it is - sorry.